oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/web/logout/PostLogoutRedirectUriProvider.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2002-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.client.oidc.web.logout;
+
+import java.net.URI;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * Strategy to create a post logout redirect URI during the OIDC logout flow.
+ *
+ * @author Sebastian Sprenger
+ * @since 5.4
+ * @see org.springframework.security.oauth2.client.oidc.web.logout.OidcClientInitiatedLogoutSuccessHandler
+ **/
+public interface PostLogoutRedirectUriProvider {
```

Comment:
I don't think we should introduce a new API for this. We may consider making `OidcClientInitiatedLogoutSuccessHandler.postLogoutRedirectUri()` `protected` OR there may be a way to leverage `URI` template variables adding to the existing `baseUrl`